src/umpire/Allocator.hpp

```diff
@@ -55,10 +55,12 @@ class Allocator {
    * \brief Free the memory at ptr.
    *
    * This method will throw an umpire::Exception if ptr was not allocated
-   * using this Allocator. If you need to deallocate memory allocated by an
-   * unknown object, use the ResourceManager::deallocate method.
+   * using this Allocator. If the pointer hasn't been used (a nullptr), 
```

Comment:
Being a nullptr, does not really indicate whether the pointer has been used.  It simply means that the pointer was set to nullptr.  I suggest the following:

```suggestion
   * using this Allocator. If the value of the pointer is set to nullptr, 
```

Reply:
Sounds good. I was trying to get the wording just right... words are hard